### PR TITLE
Add leaderelection docs

### PR DESCRIPTION
**Pull request by ChrsMark**
This PR adds docs for `kubernetes_leaderelection` provider introduced at https://github.com/elastic/beats/pull/24913

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc [hunk 1]
```diff
@@ -194,6 +194,33 @@ This means that anyone who have access to Agent Pod (`kubectl exec` for example)
...
+providers.kubernetes_leaderelection:
+  #kube_config: /Users/elastic-agent/.kube/config
+  #leader_lease: agent-k8s-leader-lock
+----
+
+`kube_config`:: (Optional) Use given config file as configuration for Kubernetes
+client. If kube_config is not set, KUBECONFIG environment variable will be
+checked and if not present it will fall back to InCluster.
+`leader_lease`:: (Optional) Specify the name of the leader lease.
+By default it is `elastic-agent-cluster-leader`.
```
Comment (exekias) [inline docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc:222]: Should we add a table with the available key from this provider?

File: docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc [hunk 2]
```diff
@@ -194,6 +194,33 @@ This means that anyone who have access to Agent Pod (`kubectl exec` for example)
...
+Kubernetes cluster like from `kube-state-metrics` endpoint.
+
+Provider needs a `kubeconfig` file so as to establish connection to Kubernetes API,
+or it can automatically reach the API if it runs in an inCluster environment (Agent runs as Pod).
+
+[source,yaml]
+----
+providers.kubernetes_leaderelection:
+  #kube_config: /Users/elastic-agent/.kube/config
+  #leader_lease: agent-k8s-leader-lock
```
Comment (exekias) [inline docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc:215]: This is not related to this PR but I have a question myself, do we have a parameter to pass the node name for `inCluster` discovery? Do we still need it in this context? In the past we got many issues related to the hostname of the Pod being different from the Pod name / node name

File: docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc (hunk 1 above)
Comment (ChrsMark) [reply docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc:222]: Nice catch! Thanks!

File: docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc (hunk 2 above)
Comment (ChrsMark) [reply docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc:215]: I don't think that we need because `inCluster` here has only to do with the k8s client initialisation, see https://github.com/elastic/beats/blob/2494ae509b1036d39bd1bbb4ec751c50347c9aff/x-pack/elastic-agent/pkg/composable/providers/kubernetesleaderelection/kubernetes_leaderelection.go#L52.

Usually `Node` setting is used by another function called `DiscoverKubernetesNode`  (see https://github.com/elastic/beats/blob/8b2cebb6229fa93aa580c6fb93dc65f988f3226c/x-pack/elastic-agent/pkg/composable/providers/kubernetes/kubernetes.go#L67) which is not used in this provider.

So I think that we are covered here?

Comment (exekias) [reply docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc:215]: Good point, I think we are fine then!

File: docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc [hunk 3]
```diff
@@ -194,6 +194,67 @@ This means that anyone who have access to Agent Pod (`kubectl exec` for example)
...
+By default it is `elastic-agent-cluster-leader`.
+
+The available key is:
+
+|===
+|Key |Type |Description
+
+|`kubernetes_leaderelection.leader`
+|`bool`
+|The value of the leadership flags. True when the Agents holds the leader lock.
```
Comment (exekias) [inline docs/en/ingest-management/elastic-agent/elastic-agent-providers.asciidoc:231]: I would rephrase a little bit:

```suggestion
|The value of the leadership flag. It is `true` when the Agent is the current leader, `false` otherwise.
```

**Review by exekias — APPROVED** [review]
